### PR TITLE
Use Semantic UI's Search component for user and repo search

### PR DESCRIPTION
**Pull request by harryxu**
Current user and repo search input is not support keyboard navigation and must use mouse click to select item.

So I use Semantic UI's Search component replace the jquery ajax code and html generating code, for improve the UI  experience.

Here is the screenshots:

**New user search**

**New repo search**

**Old user search** (Can only use mouse select):

### CODE REVIEW

File: public/js/index.js [hunk 1]
```diff
@@ -1230,104 +1230,54 @@ function hideWhenLostFocus(body, parent) {
...
-            success: function (response) {
-                var notEmpty = function (str) {
-                    return str && str.length > 0;
-                };
-
-                $results.html('');
+    $searchRepoBox.search({
+        minCharacters: 2,
+        apiSettings: {
+            url: suburl + '/api/v1/repos/search?q={query}',
```
Comment (daviian) [inline public/js/index.js]: `uid` query parameter is missing.

**Review by daviian — CHANGES_REQUESTED** [review]

File: public/js/index.js (hunk 1 above)
Comment (harryxu) [reply public/js/index.js]: @daviian  Fixed.

File: public/js/index.js [hunk 2]
```diff
@@ -1230,104 +1230,54 @@ function hideWhenLostFocus(body, parent) {
...
-    $searchUserBox.find('input').focus(function () {
-        $searchUserBox.keyup();
+        },
+        searchFields: ['login', 'full_name'],
+        showNoResults: false
     });
-    hideWhenLostFocus('#search-user-box .results', '#search-user-box');
 }
 
 // FIXME: merge common parts in two functions
```
Comment (ethantkoenig) [inline public/js/index.js]: nit: remove this comment now that it is stale

**Review by ethantkoenig — APPROVED** [review]
Looks good! Just one nit.

**Review by daviian — APPROVED** [review]

**Comment by daviian** [comment]
Please remove comment as @ethantkoenig mentioned, otherwise LGTM

**Comment by harryxu** [comment]
@ethantkoenig @daviian Done. 😃

**Comment by ethantkoenig** [comment]
LGTM

**Comment by lafriks** [comment]
Make LG-TM work again

**Pull request merged**